Drop-In/src/androidTest/java/com/braintreepayments/api/SelectPaymentMethodFragmentUITest.kt

```diff
@@ -0,0 +1,236 @@
+    }
+
+    @Test
+    fun whenStateIsRESUMED_whenVaultManagerIsEnabledAndVaultedPaymentMethodsAreLoaded_displaysVaultEditButton() {
+        val dropInRequest = DropInRequest()
+                .vaultManager(true)
+        val bundle = bundleOf("EXTRA_DROP_IN_REQUEST" to dropInRequest)
+
+        val scenario = FragmentScenario.launchInContainer(SelectPaymentMethodFragment::class.java, bundle)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        val cardNonce = CardNonce.fromJSON(JSONObject(Fixtures.PAYMENT_METHODS_VISA_CREDIT_CARD))
+        val vaultedPaymentMethods = listOf(cardNonce)
+
+        scenario.onFragment { fragment ->
+            fragment.dropInViewModel.setSupportedPaymentMethods(supportedPaymentMethods)
+            fragment.dropInViewModel.setVaultedPaymentMethods(vaultedPaymentMethods)
+        }
+
+        onView(withId(R.id.bt_vault_edit_button)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun whenStateIsRESUMED_whenVaultManagerIsDisabled_hidesVaultEditButton() {
+        val dropInRequest = DropInRequest()
+                .vaultManager(false)
+        val bundle = bundleOf("EXTRA_DROP_IN_REQUEST" to dropInRequest)
+
+        val scenario = FragmentScenario.launchInContainer(SelectPaymentMethodFragment::class.java, bundle)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        scenario.onFragment { fragment ->
+            fragment.dropInViewModel.setSupportedPaymentMethods(supportedPaymentMethods)
+        }
+
+        onView(withId(R.id.bt_vault_edit_button)).check(matches(not(isDisplayed())))
+    }
+
+    @Test
+    fun whenStateIsRESUMED_whenCreditOrDebitCardSelected_sendsPaymentMethodSelectedEvent() {
+        val dropInRequest = DropInRequest()
+                .vaultManager(false)
+        val bundle = bundleOf("EXTRA_DROP_IN_REQUEST" to dropInRequest)
+
+        val scenario = FragmentScenario.launchInContainer(SelectPaymentMethodFragment::class.java, bundle)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        scenario.onFragment { fragment ->
+            fragment.dropInViewModel.setSupportedPaymentMethods(supportedPaymentMethods)
+        }
+
+        onView(isRoot()).perform(waitFor(500))
+        onView(withText("Credit or Debit Card")).perform(click())
+        scenario.onFragment { fragment ->
+            val activity = fragment.requireActivity()
+            val fragmentManager = fragment.parentFragmentManager
+            fragmentManager.setFragmentResultListener("BRAINTREE_EVENT", activity) { requestKey, result ->
+                val event = result.get("BRAINTREE_RESULT") as SupportedPaymentMethodSelectedEvent
+                assertEquals(DropInPaymentMethodType.UNKNOWN, event.paymentMethodType)
+                countDownLatch.countDown()
+            }
+        }
+
+        countDownLatch.await()
+    }
+
+    @Test(timeout = 5000)
+    fun whenStateIsRESUMED_whenVaultManagerEnabledAndVaultedCardExists_sendsAnalyticEvent() {
```

Comment:
This is one of the Analytics assertion ideas. We basically have to collect all the events that were emitted and assert that the target analytics event is present.

Reply:
Maybe testing these can go along with a larger story to revisit Drop-in analytics events in general and align them across SDKs?

Reply:
Yeah def. Refactoring to fragment will at least organize everything but we're going to get more coverage overall. I'll make a ticket.